src/test/deploy/functions/validate.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { expect } from "chai";
+import * as fsutils from "../../../fsutils";
+import * as validate from "../../../deploy/functions/validate";
+import * as projectPath from "../../../projectPath";
+import * as FirebaseError from "../../../error";
+import * as sinon from "sinon";
+
+// have to require this because no @types/cjson available
+// tslint:disable-next-line
+const cjson = require("cjson");
+
+describe("validate", () => {
+  describe("functionsDirectoryExists", () => {
+    let sandbox: sinon.SinonSandbox;
+    let resolvePpathStub: sinon.SinonStub;
+    let dirExistsStub: sinon.SinonStub;
+
+    beforeEach(() => {
+      sandbox = sinon.sandbox.create();
+      resolvePpathStub = sandbox.stub(projectPath, "resolveProjectPath");
+      dirExistsStub = sandbox.stub(fsutils, "dirExistsSync");
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it("should not throw error if functions directory is present", () => {
+      resolvePpathStub.returns("some/path/to/project");
+      dirExistsStub.returns(true);
+      expect(() => {
```

Comment:
keep a newline between the Align portion (setting stubs) and the Act portion (actually doing the functionality). (We can't have a newline between Act and Assert since, in this case, they're one and the same.)

Reply:
Makes sense, done.